projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -71,43 +71,43 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		}
 
 		if ( $is_valid_token ) {
-			if ( 'subscribers' === $access_level ) {
-				return 'active' === $payload['blog_sub'];
-			}
-			$subscriptions = (array) $payload['subscriptions'];
-		} elseif ( is_user_logged_in() ) {
-			/*
-			 * If there is no token, but the user is logged in,
-			 * get current subscriptions and determine if the user has
-			 * a valid subscription to match the plan ID.
-			 */
+			$is_blog_subscrber  = 'active' === $payload['blog_sub'];
+			$subscriptions      = (array) $payload['subscriptions'];
+			$is_paid_subscriber = $this->validate_subscriptions( $valid_plan_ids, $subscriptions );
+		} else {
+			// Token not valid. We bail even unless the post can be accessed publicly.
+			return 'everybody' === $access_level;
+		}
 
-			/**
-			 * Filter the subscriptions attached to a specific user on a given site.
-			 *
-			 * @since 9.4.0
-			 *
-			 * @param array $subscriptions Array of subscriptions.
-			 * @param int   $user_id The user's ID.
-			 * @param int   $site_id ID of the current site.
-			 */
-			$subscriptions = apply_filters(
-				'earn_get_user_subscriptions_for_site_id',
-				array(),
-				wp_get_current_user()->ID,
-				$this->get_site_id()
-			);
+		return self::user_has_access( $access_level, $is_blog_subscrber, $is_paid_subscriber );
+	}
 
-			if ( empty( $subscriptions ) ) {
-				return false;
-			}
-			// format the subscriptions so that they can be validated.
-			$subscriptions = self::abbreviate_subscriptions( $subscriptions );
-		} else {
-			return false;
+	/**
+	 * Return if the user has access to the content depending on the access level and the user rights
+	 *
+	 * @param string $access_level Post or blog access level.
+	 * @param bool   $is_blog_subscrber Is user a subscriber of the blog.
+	 * @param bool   $is_paid_subscriber Is user a paidsubscriber of the blog.
+	 *
+	 * @return bool does the user have access to the post/blog.
+	 */
+	protected static function user_has_access( $access_level, $is_blog_subscrber, $is_paid_subscriber ) {
```

Comment:
Yes fantastic, this method needs to be broken out 👍

Reply:
Are we in the most base class? We should have this in the most base class in this inheritance chain

Reply:
Yes, all the parents are interfaces